spec/logstash/size_queue_timeout_spec.rb

```diff
@@ -0,0 +1,68 @@
+require "spec_helper"
+require "logstash/sized_queue_timeout"
+require "flores/random"
+
+describe "LogStash::SizedQueueTimeout" do
+  let(:max_size) { Flores::Random.integer(2..100) }
+  let(:element) { Flores::Random.text(0..100) }
+  # let(:max_size) { 3 }
+  # let(:element) { "Hello Logstash" }
+
+  subject { LogStash::SizedQueueTimeout.new(max_size) }
+  
+  it "adds element to the queue" do
+    subject << element
+    expect(subject.size).to eq(1)
+  end
+
+  it "allow to pop element from the queue" do
+    subject << element
+    subject << "awesome"
+
+    expect(subject.pop_no_timeout).to eq(element)
+  end
+  
+  context "when the queue is full" do
+    before do
+      max_size.times do
+        subject << element
+      end
+    end
+
+    it "block with a timeout" do
+      expect {
+        subject << element
+      }.to raise_error(LogStash::SizedQueueTimeout::TimeoutError)
+    end
+
+    it "unblock when we pop" do
+      blocked = testing_thread do
+        subject << element
+      end
+
+      expect(blocked.status).to eq("sleep")
```

Comment:
Weird, this means taht rspec's failure exception doesn't inherit from StandardError, so `Stud::Try` never catches it (https://github.com/jordansissel/ruby-stud/blob/master/lib/stud/try.rb#L95)

I can update Stud::Try to catch all exceptions instead of just the family of StandardError. Thoughts?

If I do that, then the Stud::try(5.times) will fail a few times, sleep each time (exponential backoff) hoping that the next attempt will succeed and not raise an error.